WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Service/BlogDashboardService.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+import WordPressKit
+
+class BlogDashboardService {
+    let remoteService: DashboardServiceRemote
+
+    init(managedObjectContext: NSManagedObjectContext, remoteService: DashboardServiceRemote? = nil) {
+        self.remoteService = remoteService ?? DashboardServiceRemote(wordPressComRestApi: WordPressComRestApi.defaultApi(in: managedObjectContext, localeKey: WordPressComRestApi.LocaleKeyV2))
+    }
+
+    func fetch(wpComID: Int, completion: @escaping (DashboardSnapshot) -> Void) {
+        let cardsToFetch: [String] = DashboardCard.allCases
+            .filter { $0.isRemote }
+            .map { $0.rawValue }
+
+        remoteService.fetch(cards: cardsToFetch, forBlogID: wpComID, success: { [weak self] cards in
+
+            var snapshot = DashboardSnapshot()
+
+            DashboardCard.allCases.forEach { card in
```

Comment:
Done in e2cd152ca4c23a1da7446a75e2db583c45b6a581

Reply:
Thanks! What do you think about the second part of my comment?  I was thinking we could iterate through remote cases / local cases individually, instead of looping through all cases then checking if it's remote or not

Reply:
Oh, I missed that part, sorry.

The only I see is that this will create two groups: local cards will be displayed first, remote later. This is implicit while displaying with the `DashboardCard` order seems more natural to me. Wdyt?

Reply:
Right, I forgot about the ordering. Using the implicit ordering on DashboardCard sounds good.